File: app/src/main/java/com/example/williamburbatt/androidstudioapp5/MainActivity.java
package com.example.williamburbatt.androidstudioapp5;

import android.content.Intent;
import android.graphics.drawable.ColorDrawable;
import android.net.Uri;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.ListAdapter;
import android.widget.ListView;
import android.widget.PopupWindow;
import android.widget.TextView;
import android.widget.Toast;

import org.w3c.dom.Text;

//dont change this... crashes too much
public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        String [] attraction = {"Name","Date of Birth","Address","Placeholder One","Placeholder Two"};
        //Need the adapter and to tell it what kind of list it's going into
        ArrayAdapter<String> arr = new ArrayAdapter<String>(this, R.layout.activity_listview,attraction);
        //put them all into listview
        ListView lv = (ListView)findViewById(R.id.list1);
        lv.setAdapter(arr);
        lv.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> a, View v, int position,
                                    long id) {
                switch (position) {
                    case 0:
                      /**  //this tests new page passing position
                        Intent intent = new Intent(MainActivity.this, test2.class);
                        intent.putExtra("pos", position);
                        startActivity(intent);**/
                        popUp("Edit Name");
                        break;
                    case 1:
                        //this tests linking to a website
                        popUp("Edit Date of Birth");
                        break;
                    case 2:
                        popUp("Edit Address");
                        break;
                    case 3:
                        Toast.makeText(MainActivity.this, "placeholder: " + position, Toast.LENGTH_LONG).show();
                        break;
                    case 4:
                        Toast.makeText(MainActivity.this, "placeholder: " + position, Toast.LENGTH_LONG).show();
                        break;
                }
            }
        });
        backButton();
    }
    public void backButton(){
        ImageView back = (ImageView) findViewById(R.id.imageView);
        back.setOnClickListener(new View.OnClickListener(){

            @Override
            public void onClick(View v) {
                Toast.makeText(MainActivity.this, "To be implemented later ", Toast.LENGTH_LONG).show();
            }
        });
    }

    public void popUp(String s){

        LayoutInflater inflater = (LayoutInflater) MainActivity.this.getSystemService(MainActivity.LAYOUT_INFLATER_SERVICE);
//Inflate the view from a predefined XML layout (no need for root id, using entire layout)
        View layout = inflater.inflate(R.layout.layout_custom_dialog,null);
        float density=MainActivity.this.getResources().getDisplayMetrics().density;
        final PopupWindow pw = new PopupWindow(layout, (int)density*480, (int)density*280, true);
        ((TextView) layout.findViewById(R.id.h1)).setText(s);
        //Button to close the pop-up
        ((Button) layout.findViewById(R.id.button)).setOnClickListener(new View.OnClickListener() {
            public void onClick(View v) {
                pw.dismiss();
            }
        });
        //Set up touch closing outside of pop-up
        pw.setBackgroundDrawable(new ColorDrawable(android.graphics.Color.TRANSPARENT));

        // display the pop-up in the center
        pw.showAtLocation(layout, Gravity.CENTER, 0, 0);
    }
    }

File: app/src/main/java/com/example/williamburbatt/androidstudioapp5/MainActivity.kt
package com.example.williamburbatt.androidstudioapp5


import android.accounts.Account
import android.content.Context
import android.graphics.drawable.ColorDrawable
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log
import android.view.Gravity
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.*

//This page was created by William A Burbatt 4/24/18
//dont change this... crashes too much
class MainActivity : AppCompatActivity() {

    private var listAccounts = ArrayList<MyAccount>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        listAccounts.add(MyAccount(1, "JavaSampleApproach", "Java technology, Spring Framework - approach to Java by Sample."))
        listAccounts.add(MyAccount(2, "Kotlin Android Tutorial", "Create tutorial for people to learn Kotlin Android. Kotlin is now an official language on Android. It's expressive, concise, and powerful. Best of all, it's interoperable with our existing Android languages and runtime."))
        listAccounts.add(MyAccount(3, "Android Studio", "Android Studio 3.0 provides helpful tools to help you start using Kotlin. Convert entire Java files or convert code snippets on the fly when you paste Java code into a Kotlin file."))
        listAccounts.add(MyAccount(4, "Java Android Tutorial", "Create tutorial for people to learn Java Android. Learn Java in a greatly improved learning environment with more lessons, real practice opportunity, and community support."))
        listAccounts.add(MyAccount(5, "Spring Boot Tutorial", "Spring Boot help build stand-alone, production Spring Applications easily, less configuration then rapidly start new projects."))

        var accAdapter = AccountAdapter(this, listAccounts)
        val lv = findViewById<View>(R.id.list1) as ListView

        lv.adapter = accAdapter
        lv.onItemClickListener = AdapterView.OnItemClickListener { adapterView, view, position, id ->
            Toast.makeText(this, "Click on " + listAccounts[position].title, Toast.LENGTH_SHORT).show()
        }
    }



    inner class AccountAdapter : BaseAdapter {

        private var accList = ArrayList<MyAccount>()
        private var context: Context? = null

        constructor(context: Context, notesList: ArrayList<MyAccount>) : super() {
            this.accList = notesList
            this.context = context
        }

        override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View? {

            val view: View?
            val vh: ViewHolder

            if (convertView == null) {
                view = layoutInflater.inflate(R.layout.activity_listview, parent, false)
                vh = ViewHolder(view)
                view.tag = vh
            } else {
                view = convertView
                vh = view.tag as ViewHolder
            }

            vh.accountTitle.text = accList[position].title
            vh.accountContent.text = accList[position].content

            return view
        }

        override fun getItem(position: Int): Any {
            return accList[position]
        }

        override fun getItemId(position: Int): Long {
            return position.toLong()
        }

        override fun getCount(): Int {
            return accList.size
        }
    }

    private class ViewHolder(view: View?) {
        val accountTitle: TextView = view?.findViewById(R.id.label) as TextView
        val accountContent: TextView = view?.findViewById(R.id.content) as TextView

        //  if you target API 26, you should change to:
//        init {
//            this.tvTitle = view?.findViewById<TextView>(R.id.tvTitle) as TextView
//            this.tvContent = view?.findViewById<TextView>(R.id.tvContent) as TextView
//        }
    }
}
    class MyAccount {

        var id: Int? = null
        var title: String? = null
        var content: String? = null

        constructor(id: Int, title: String, content: String) {
            this.id = id
            this.title = title
            this.content = content
        }

    }
